network/server.go

```diff
@@ -377,14 +384,35 @@ func (s *Server) hasPeer(peerID peer.ID) bool {
 }
 
 func (s *Server) numOpenSlots() int64 {
-	n := int64(s.config.MaxPeers) - (s.numPeers() + s.identity.numPending())
+	n := s.maxOutboundConns() - s.outboundConns()
 	if n < 0 {
 		n = 0
 	}
 
 	return n
 }
 
+func (s *Server) inboundConns() int64 {
+	count := atomic.LoadInt64(&s.inboundConnCount)
+	if count < 0 {
+		count = 0
+	}
```

Comment:
Why would you need to check if count < 0? In what scenario would you have it < 0?